pemFioi/blocklyRobot_lib-1.0.0.js

```diff
@@ -3301,7 +3301,42 @@ var getContext = function(display, infos, curLevel) {
          resetItems();
       }
    };
-   
+
+   context.resetDisplay = function() {
+      if(context.display) {
+         this.raphaelFactory.destroyAll();
+         if(paper !== undefined)
+            paper.remove();
+         paper = this.raphaelFactory.create("paperMain", "grid", infos.cellSide * context.nbCols * scale, infos.cellSide * context.nbRows * scale);
+         resetBoard();
+         redisplayAllItems();
+         context.updateScale();
+         $("#nbMoves").html(context.nbMoves);
+      }
+   }
+
+   context.getCurrentState = function() {

```

Comment:
I would name it `getInnerState` or `getContextState`, and rename `reloadState` to `reloadInnerState` or `reloadContextState`, to make the code clearer (as `reloadState `is the name of a Bebras API point, which spans calls up to `quickAlgo/subtask.js`, and naming it the same would be confusing).